### PR TITLE
Fixing differential rotation of coordinates again.

### PR DESCRIPTION
**Pull request by wafels**
### Description

Fixes the differential rotation of coordinates again.  With this PR, the `solar_rotate_coordinate` function returns a coordinate that accounts for the location of the `new_observer`.

Fixes #<Issue Number>

### CODE REVIEW

File: sunpy/physics/differential_rotation.py [hunk 1]
```diff
@@ -246,14 +248,17 @@ def solar_rotate_coordinate(coordinate, observer=None, time=None, **diff_rot_kwa
     drot = diff_rot(interval, heliographic_coordinate.lat.to(u.degree), **diff_rot_kwargs)
 
     # Rotate the input co-ordinate as seen by the original observer
-    heliographic_rotated = SkyCoord(heliographic_coordinate.lon + drot, heliographic_coordinate.lat,
-                                    obstime=coordinate.obstime, observer=coordinate.observer,
-                                    frame=HeliographicStonyhurst)
+    heliographic_rotated = SkyCoord(heliographic_coordinate.lon + drot,
+                                    heliographic_coordinate.lat,
+                                    heliographic_coordinate.radius,
+                                    obstime = new_observer.obstime,
```
Comment (Cadair) [inline sunpy/physics/differential_rotation.py]: no spaces around kwarg  `=` please.

**Comment by Cadair** [comment]
@nabobalis backport please.

**Comment by wafels** [comment]
This and the warping code in #2972 can be used to reproject images to different viewpoints.

**Comment by nabobalis** [comment]
Can I just protest about the history?

**Comment by Cadair** [comment]
Me thinks we might have to give @wafels a lesson on rewriting git history :stuck_out_tongue:

**Pull request merged**